core/src/scene/importer.cpp

```diff
@@ -0,0 +1,257 @@
+                        auto uniformValue = uniformNode.second;
+                        if (SceneLoader::parseStyleUniforms(uniformValue, nullptr, styleUniform) &&
+                                styleUniform.type == "sampler2D") {
+                            if (styleUniform.value.is<UniformTextureArray>()) {
+                                setNormalizedTexture(uniformValue,
+                                             styleUniform.value.get<UniformTextureArray>().names,
+                                             parentPath);
+                            } else {
+                                auto name = styleUniform.value.get<std::string>();
+                                setNormalizedTexture(uniformValue, {name}, parentPath);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+bool Importer::loadScene(const std::string& path) {
+
+    auto scenePath = path;
+    auto sceneString = stringFromFile(scenePath.c_str(), PathType::resource);
+    auto sceneName = getFilename(scenePath);
+
+    if (m_scenes.find(sceneName) != m_scenes.end()) { return true; }
+
+    // Make sure all references from uber scene file are relative to itself, instead of being
+    // absolute paths (Example: when loading a file using command line args).
+    // TODO: Could be made better later.
+    if (m_scenes.size() == 0 && scenePath[0] == '/') { scenePath = getFilename(scenePath); }
+
+    try {
+        auto root = YAML::Load(sceneString);
+        normalizeSceneImports(root, scenePath);
+        normalizeSceneTextures(root, scenePath);
+        auto imports = getScenesToImport(root);
+        m_scenes[sceneName] = root;
+        for (const auto& import : imports) {
+            // TODO: What happens when parsing fails for an import
+            loadScene(import);
+        }
+    }
+    catch (YAML::ParserException e) {
+        LOGE("Parsing scene config '%s'", e.what());
+        return false;
+    }
+    return true;
+}
+
+std::vector<std::string> Importer::getScenesToImport(const Node& scene) {
+
+    std::vector<std::string> sceneNames;
+
+    if (const Node& import = scene["import"]) {
+        if (import.IsScalar()) {
+            sceneNames.push_back(import.Scalar());
+        }
+        else if (import.IsSequence()) {
+            for (const auto& imp : import) {
+                if (imp.IsScalar()) { sceneNames.push_back(imp.Scalar()); }
+            }
+        }
+    }
+
+    return sceneNames;
+}
+
+std::vector<std::string> Importer::getImportOrder() {
+
+    std::vector<std::pair<std::string, std::string>> dependencies;
+
+    for (const auto& scene : m_scenes) {
+        const auto& name = scene.first.k;
+        const auto& sceneRoot = scene.second;
+        for (const auto& import : getScenesToImport(sceneRoot)) {
+            auto importName = getFilename(import);
+            // TODO: fixme do not allow cycles
```

Comment:
The current implementation seems to do Very Bad Things when there is an import cycle. There's clearly not one 'correct' way to resolve a cycle, but we definitely can't crash. 

When `topologicalSort` encounters a cycle, it treats it as an error in the inputs and returns an empty list. You can check for this result and respond appropriately - for instance, using only the root scene and logging an error.